facilities/serializers.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework import serializers
+from .models import Owner, Facility, Service
+from common.serializers import AbstractFieldsMixin
+
+
+class OwnerSerializer(AbstractFieldsMixin, serializers.ModelSerializer):
+    class Meta:
+        model = Owner
+
+
+class ServiceSerializer(AbstractFieldsMixin, serializers.ModelSerializer):
+    class Meta:
+        model = Service
+
+
+class FacilityReadSerializer(
+        AbstractFieldsMixin, serializers.ModelSerializer):
+
+    class Meta:
+        model = Facility
+
+
+class FacilitySerializer(
+        AbstractFieldsMixin, serializers.ModelSerializer):
+    services = ServiceSerializer(many=True, required=False)
+
+    def create(self, validated_data):
```

Comment:
This is good code; but it will be simpler if we keep the API to simple, reliable CRUD; @liyosi and @owagaantony can create front-end wrappers that do this stuff that you are doing here.

i.e. unless there is a direct request and a good reason, let M2M fields be manipulated through the join tables.

For the front-enders: @owagaantony @sombo09 @liyosi  - design the data input forms so that _it is natural to save the "header" information first_.